crates/oxc_linter/src/rules/eslint/no_console.rs

```diff
@@ -78,8 +81,12 @@ impl Rule for NoConsole {
                             .iter()
                             .any(|s| mem.static_property_name().is_some_and(|f| f == s))
                     {
-                        if let Some(mem) = mem.static_property_info() {
-                            ctx.diagnostic(no_console_diagnostic(mem.0));
+                        if let Some((mem_span, _)) = mem.static_property_info() {
+                            let diagnostic_span = ident.span().merge(&mem_span);
+                            ctx.diagnostic_with_suggestion(
+                                no_console_diagnostic(diagnostic_span),
+                                |fixer| remove_console(fixer, ctx, node),
```

Comment:
Seems we also need to port https://github.com/eslint/eslint/blob/1452dc9f12c45c05d7c569f737221f0d988ecef1/lib/rules/no-console.js#L115-L130, or it will cause dangerous fix.

```
foo()
console.log();
[1, 2, 3].forEach(a => doSomething(a))
```

Reply:
This function seems useful in other cases besides no-console. Maybe we move it to utils, or even into the fixer itself?